pkg/controller/controller_utils.go

```diff
@@ -402,3 +405,129 @@ func ClusterAPIProviderStatusFromClusterStatus(clusterStatus *clusteroperator.Cl
 		Raw: buffer.Bytes(),
 	}, nil
 }
+
+// MachineSetSpecFromClusterAPIMachineSpec gets the cluster-operator MachineSetSpec from the
+// specified cluster-api MachineSet.
+func MachineSetSpecFromClusterAPIMachineSpec(ms *clusterapi.MachineSpec) (*clusteroperator.MachineSetSpec, error) {
+	if ms.ProviderConfig.Value == nil {
+		return nil, fmt.Errorf("No Value in ProviderConfig")
+	}
+	obj, _, err := api.Codecs.UniversalDecoder(clusteroperator.SchemeGroupVersion).Decode([]byte(ms.ProviderConfig.Value.Raw), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	spec, ok := obj.(*clusteroperator.MachineSetProviderConfigSpec)
+	if !ok {
+		return nil, fmt.Errorf("Unexpected object: %#v", obj)
```

Comment:
As in `clusterOperatorMachineSetSpec` in `pkg/clusterapi/aws/actuator.go`, let's use the `GroupVersionKind` instead of dumping the object.

Reply:
Updating this here and for a couple other similar methods in this package.